src/Contacts.ts

```diff
@@ -0,0 +1,211 @@
+import Client from './Client'
+import { privatePreferences } from '@xmtp/proto'
+import { EnvelopeWithMessage, buildUserPrivatePreferencesTopic } from './utils'
+
+export type AllowListPermissionType = 'allow' | 'block' | 'unknown'
+
+export type AllowListEntryType = 'address'
+
+export class AllowListEntry {
```

Comment:
I'm not the biggest fan of this being `AllowList....` etc From the proposal it encourages the verbage `ConsentList` `ConsentState` etc... I went around the iOS Android and RN sdks to clean this up to be more consistent with the word Consent instead of Allow since it returns block, allow, and unknown.